### ISSUE TITLE
[BUG] User warning in losses computation

### ISSUE THREAD

**Issue by banctilrobitaille** [sami-ets/SAMITorch#23]
**Describe the bug**
UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor). requires_grad=False, dtype=torch.float)

**To Reproduce**
Run the losses tests

**Expected behavior**
No user warning :smile_cat: 

**Screenshots**
![image](https://user-images.githubusercontent.com/5373064/61889808-1cbda080-aed4-11e9-8571-1febd1a2fa0f.png)

**Desktop (please complete the following information):**
 - OS platform and distribution: Linux Ubuntu 18.04.2 LTS
 - Python version:Python 3.7
 - CUDA/cuDNN version:CUDA 10.0
 - GPU model and memory: NVIDIA GeForce 1080 8 GB

**Additional context**
